File: Leetcode/group_anagrams.py

class Solution:
	def groupAnagrams(self, strs):
		pass

	def isAnagram(self, str1, str2):
		if len(str1) != len(str2): return False

		counter = [0 for r in range(200)]
		
		for r in range(len(str1)):
			counter[ord(str1[r])] += 1
			counter[ord(str2[r])] -= 1

		for r in range(200):
			if counter[r] != 0:
				return False

		return True

File: Leetcode/group_anagrams.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution{
public:
	vector<vector<string>> groupAnagrams(vector<string>& strs) {
		unordered_map<string, vector<string>> mp;

		for(string str: strs){
			string word = str;
			sort(word.begin(), word.end());
			mp[word].push_back(str);
		}

		vector<vector<string>> ans;
		for(auto group: mp){
			ans.push_back(group.second);
		}
		return ans;
	}

};


int main(){

	Solution sol;
	vector<string> strs {"eat", "tea", "tan", "ate", "nat", "bat"};
	vector<vector<string>> groups = sol.groupAnagrams(strs);
	for (vector<string> group: groups){
		for (string str : group){
			cout << str << ' ';
		}cout << endl;
	}

	return 0;
}

